plugin/cmd/kube-scheduler/app/server.go

```diff
@@ -114,7 +114,7 @@ func Run(s *options.SchedulerServer) error {
 	rl := &resourcelock.EndpointsLock{
 		EndpointsMeta: metav1.ObjectMeta{
 			Namespace: "kube-system",
-			Name:      "kube-scheduler",
+			Name:      s.SchedulerName,
```

Comment:
I would encourage namespaced names like that to avoid collisions... that's why I think making that serve double-duty as the endpoint lock object name is a bad idea...